Framework/Algorithms/test/CreateWorkspaceTest.h

```diff
@@ -3,10 +3,53 @@
 #include "MantidAPI/TextAxis.h"
 #include "MantidAlgorithms/CreateWorkspace.h"
 #include "MantidKernel/Memory.h"
+#include "MantidTestHelpers/ParallelRunner.h"
 #include "MantidTestHelpers/WorkspaceCreationHelper.h"
 
+using namespace Mantid;
 using namespace Mantid::API;
 
+namespace {
+void run_create(const Parallel::Communicator &comm,
+                const std::string &storageMode) {
+  Mantid::Algorithms::CreateWorkspace alg;
+  alg.setCommunicator(comm);
+  alg.initialize();
+  std::vector<double> dataEYX(2000);
+  int nspec = 1000;
+  alg.setProperty<int>("NSpec", nspec);
+  alg.setProperty<std::vector<double>>("DataX", dataEYX);
+  alg.setProperty<std::vector<double>>("DataY", dataEYX);
+  alg.setProperty<std::vector<double>>("DataE", dataEYX);
+  // In a true MPI run we could simply use "out", but in the threaded
+  // fake-runner we have only one ADS, so we have to avoid clashes.
+  std::string outName("out" + std::to_string(comm.rank()));
```

Comment:
Code like this now shows up in every unit test of parallel execution. I would really like to avoid this. I have considered running the algorithms as a child algorithm since then I do not have to set a name myself. However, we then need a parent algorithm.

Solutions?
- Create a test helper `ParallelAlgorithmFactory`, which holds a parent algorithm and can spawn the testee as a child.
- other?